semeio/workflows/localisation/local_script_lib.py

```diff
@@ -286,22 +539,86 @@ def add_ministeps(
                 )
             elif impl_type == ErtImplType.FIELD:
                 if corr_spec.field_scale is not None:
-                    field_config = node.getFieldModelConfig()
-                    check_if_ref_point_in_grid(corr_spec.ref_point, grid_for_field)
-                    activate_and_scale_correlation(
-                        grid_for_field,
-                        corr_spec.field_scale.method,
-                        model_param_group.row_scaling(node_name),
-                        corr_spec.ref_point,
-                        field_config.get_data_size(),
-                        corr_spec.field_scale.main_range,
-                        corr_spec.field_scale.perp_range,
-                        corr_spec.field_scale.azimuth,
+                    debug_print(
+                        "Scale correlations using method: "
+                        f"{corr_spec.field_scale.method}",
+                        LogLevel.LEVEL3,
                         user_config.log_level,
                     )
+                    field_config = node.getFieldModelConfig()
+                    row_scaling = model_param_group.row_scaling(node_name)
+                    data_size = grid_for_field.get_num_active()
+                    data_size2 = field_config.get_data_size()
+                    assert data_size == data_size2
+                    param_for_field = None
+                    if corr_spec.field_scale.method == "gaussian_decay":
```

Comment:
A lot of duplication with `exponential_decay`, can it be collapsed?

Reply:
exponential_decay and gaussian_decay have the same type of input. I can replace the 
**if corr_spec.field_scale_method == "gaussian_decay":**
with
**if corr_spec.field_scale_method in decay_function_list:**
and define the decay_function_list as list of all method names using the same type of input for decay functions
and within that select a more generic apply_decay function with method name as parameter and within the apply_decay function put the if statements distinguishing between the decay functions with the same type of input. ? Something like that?